### PR TITLE
Fix a Py2/3 incompatibility in profiler.

### PR DESCRIPTION
**Pull request by tvalentyn**
Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by tvalentyn** [comment]
R: @Hannah-Jiang

**Comment by Hannah-Jiang** [comment]
LGTM

File: sdks/python/apache_beam/utils/profiler.py [hunk 1]
```diff
@@ -88,7 +88,11 @@ def __exit__(self, *args):
       self.profile_output = dump_location
 
     if self.log_results:
-      s = io.StringIO()
+      try:
+        import StringIO  # Python 2
```
Comment (Hannah-Jiang) [inline sdks/python/apache_beam/utils/profiler.py]: NIT: Do we want to add a comment something like `# TODO(BEAM-8371): Python2 should use StringIO` so it's easy to find it when we cleanup Py2 -> Py3?

Comment (tvalentyn) [reply sdks/python/apache_beam/utils/profiler.py]: There are several keywords we'll have to look for during this cleanup. "Python 2" should be one of them, that's why I added it. Tracking all possible keywords to search in https://issues.apache.org/jira/browse/BEAM-7372.

**Pull request merged**